internal/teststep/config.go

```diff
@@ -126,6 +127,37 @@ func Configuration(req ConfigurationRequest) Config {
 	return nil
 }
 
+// ConfigString is used to hold string representations of
+// Terraform configuration.
+type ConfigString string
+
+// AddTerraformBlock inspects ConfigString, and prefixes
+// with an empty terraform block providing the configuration
+// string is not JSON, and a Terraform block is not already
+// present.
+func (c ConfigString) AddTerraformBlock() string {
+	if !c.IsJSON() && !c.HasTerraformBlock() {
+		str := strings.TrimLeft(string(c), "\n")
+
+		return fmt.Sprintf("terraform {}\n\n%s", str)
```

Comment:
For this to work as expected, the `terraform` configuration in this case must include a correct `required_providers` definition to match the provider being tested, e.g. 

```terraform
terraform {
  required_providers {
    NAME = {
      source = "ADDRESS"
    }
  }
}
```

This is what future Terraform provider functionality will require in the configuration so Terraform explicitly knows which provider is being referenced in a configuration (rather than implicitly guessing any missing provider declaration is in the hashicorp namespace). Here's some internal background about this problem: https://github.com/hashicorp/terraform-proposals/issues/66

There is _somewhat_ of an example of this in `(TestStep).providerConfig()` but its only wired to the `ExternalProviders` information. This implementation needs to account for a lot more, unfortunately:

I believe the NAME values must contain all the map keys of all the following defined on the `TestCase`/`TestStep`:

* `ExternalProviders`
* `ProtoV5ProviderFactories`
* `ProtoV6ProviderFactories`

(We don't necessarily need to include `Providers`/`ProviderFactories` as terraform-plugin-sdk/v2 providers will never receive the future enhancements needing the required provider configuration.)

And the ADDRESS value _may_ also need to depend on things like the [`TF_ACC_PROVIDER_NAMESPACE` environment variable](https://developer.hashicorp.com/terraform/plugin/testing/acceptance-tests#environment-variables). We might need to introduce some additional testing with non-hashicorp namespace providers and the `ProtoV{5,6}ProviderFactory` fields to verify whether that value must be consulted or not.